### DISCUSSION TITLE
Modifying the tutorial codes from RACCOON

### DISCUSSION THREAD

**Discussion by NeedOfHelp** [idaholab/moose#23814] [Q&A Modules: Phase field] [answered]
Hi

I am a new user of Moose and RACCOON (A parallel finite-element code specialized in phase-field for fracture) and I am currently making simulations of crack and crack growth using phase field. I am using the tutorial code from RACCOON that is named "Tutorial 2: Mode-I crack propagation" where the full code can be found in "tutorials/mode1_brittle_fracture/elasticity.i". In that code the package 
[elasticity]
    type = SmallDeformationIsotropicElasticity
is used, and I'm wondering if it is possible to modify it to Anisotropic elasticity instead?
If so, what is the other module called? and how does it affect the rest of the code? and how do I apply it?

I have looked through the library of RACOON but didn't find anything "anisotropy" related. 

I appreciate any help I can get.

**Comment by GiudGiud** [comment]
@hugary1995 will know more than me but to go from IsotropicElasticity to an anisotropic model, you just need to find the model that does not specify that it s isotropic.

So here, maybe look for SmallDeformationElasticity, and if it doesn't exist, look for a similar model in the phase field module

**Reply by NeedOfHelp** [reply]
Thanks for your relpy.
I found this, https://mooseframework.inl.gov/source/materials/ComputeElasticityTensor.html , where it shows how to compute different kinds of elasticity tensors. The problem is that I am new to moose and phase field in general so when I tried to modify the codes "elasticity.i" and "fracture.i", it resulted in bunch of errors since changing/adding in one part of the code, you need to change/add something else in another part of the code.

**Comment by hugary1995** [comment, marked as answer]
RACCOON currently doesn't support anisotropic elasticity.
- Bad news: You will have to write code to do that, which requires some basic knowledge of C++.
- Good news: It will be a straightforward extension.
- Bad news: I currently don't have the bandwidth to do that for you. So you are on your own, though I'd be happy to help along the way.

So, if you accept the challenge, show me your equations first, and then I can tell you which files to modify/adapt.

**Reply by NeedOfHelp** [reply]
Thank you so much for your assistance!

I have no background in C++ and I can only program in python and matlab, so it will be difficult for me to implement a new code. So instead I'm wondering if it is possible to modify the mode1_brittle_fracture code so it instead applies cyclic loading?

Or maybe change the angle on how the force is applied  in order to see how the fracture changes?

Are those things possible in RACCOON?

**Reply by hugary1995** [reply]
Yes, these are possible. In fact, these are regular tasks handled by MOOSE, so I recommend you to go through the MOOSE tutorials first.

**Reply by NeedOfHelp** [reply]
Thanks, will do.

**Reply by d5sneha** [reply]
what is the update on anisotropy in raccoon?
